src/ProcessMaker/Nayra/Bpmn/StateTrait.php

```diff
@@ -43,27 +44,32 @@
      * @param EntityInterface $owner
      * @param string $name
      */
-    protected function initState(EntityInterface $owner, $name)
+    protected function initState(FlowNodeInterface $owner, $name)
     {
         $this->tokens = new Collection();
         $this->setFactory($owner->getFactory());
         $this->setName($name);
+        $owner->addState($this);
     }
 
     /**
      * Consume a token in the current state.
      *
+     * @param TokenInterface $token
      * @return bool
+     *
+     * @throws UnableConsumeTokenException
      */
     public function consumeToken(TokenInterface $token)
     {
         $tokenIndex = $this->tokens->indexOf($token);
         $valid = $tokenIndex !== false;
-        if ($valid) {
-            $this->tokens->splice($tokenIndex, 1);
-            $this->notifyEvent(StateInterface::EVENT_TOKEN_CONSUMED, $token);
-            $token->getInstance()->removeToken($token);
+        if (!$valid) {
```

Comment:
With this code this function returns always true, otherwise it throws an exception. Do this make sense?

Reply:
No, it should return false. I will restore the right logic.